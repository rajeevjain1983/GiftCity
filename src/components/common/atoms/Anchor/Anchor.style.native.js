// 9fbef606107a605d69c0edbcd8029e5d 
import styled, { css } from 'styled-components/native';

export const AnchorStyles = css`
  font-family: ${props => props.theme.typography.fonts.secondary};
  ${props =>
    props.anchorVariation === 'custom'
      ? `
            color: ${
              props.colorName
                ? props.theme.colorPalette[props.colorName]
                : props.theme.colors.ANCHOR.PRIMARY
            };
            margin: ${props.margins ? props.margins : '0 0 0 0'};
          `
      : ''};
  ${props =>
    props.anchorVariation === 'primary'
      ? `
              color: ${props.theme.colors.ANCHOR.PRIMARY};
            `
      : ''};
  ${props =>
    props.anchorVariation === 'secondary'
      ? `
                  color: ${props.theme.colors.ANCHOR.SECONDARY};
                `
      : ''};
  ${props =>
    props.anchorVariation === 'tertiary'
      ? `
                  color: ${props.theme.colors.ANCHOR.TERTIARY};
                `
      : ''};
  ${props =>
    props.anchorVariation === 'grayed'
      ? `
                  color: ${props.theme.colors.ANCHOR.GRAYED};
                  `
      : ''};
  ${props =>
    props.anchorVariation === 'white'
      ? `
                  color: ${props.theme.colors.WHITE};
                `
      : ''};
  ${props =>
    props.fontSizeVariation === 'small'
      ? `
                  font-size: ${props.theme.fonts.fontSize.anchor.small}px;
                  `
      : ''};
  ${props =>
    props.fontSizeVariation === 'medium'
      ? `
                  font-size: ${props.theme.fonts.fontSize.anchor.medium}px;
                  `
      : ''};
  ${props =>
    props.fontSizeVariation === 'large'
      ? `
                  font-size: ${props.theme.fonts.fontSize.anchor.large}px;
                      `
      : ''};
  ${props =>
    props.fontSizeVariation === 'xlarge'
      ? `
                  font-size: ${props.theme.fonts.fontSize.anchor.xlarge}px;
                    `
      : ''};
  ${props =>
    props.fontWeightVariation === 'active'
      ? `
                      font-weight: ${props.theme.fonts.fontWeight.bold};
                    `
      : ''};
  ${props =>
    props.fontWeightVariation === 'extrabold'
      ? `
                      font-weight: ${props.theme.typography.fontWeights.extrabold};
                    `
      : ''};
  ${props =>
    props.fullWidth
      ? `
                      width: 100%;
                      `
      : ''};
  ${props =>
    props.underline
      ? `
                      text-decoration: underline;

                        `
      : 'text-decoration: none;'};
  ${props =>
    props.underlineBlue
      ? `             text-decoration: underline;
                      text-decoration-color: ${props.theme.colors.ANCHOR.SECONDARY};

                        `
      : ''};
  ${props =>
    props.underlineWhite
      ? `             text-decoration: underline;
                      text-decoration-color: ${props.theme.colorPalette.white};

                        `
      : ''};

  ${props =>
    props.withCaret
      ? `
                        &:after {
                        content: "â€º";
                        margin-left: 5px;
              }
            `
      : ''};
  ${props =>
    props.centered
      ? `
                     text-align: center;
                      `
      : ''};
  ${props => (props.inheritedStyles ? props.inheritedStyles : '')};
`;

export const AnchorView = styled.TouchableOpacity`
  align-items: center;
  display: flex;
  flex-direction: row;
  justify-content: ${props => props.justifyContent || `center`};
`;

export const AnchorIcon = styled.Image`
  width: 3.7px;
  height: 7px;
  margin-top: ${props => props.theme.spacing.ELEM_SPACING.XXXS};
  margin-left: ${props => props.theme.spacing.ELEM_SPACING.XS};
`;

export default {
  AnchorStyles,
  AnchorView,
  AnchorIcon,
};
