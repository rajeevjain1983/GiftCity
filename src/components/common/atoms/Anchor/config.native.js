// 9fbef606107a605d69c0edbcd8029e5d 
/**
 * @description - Global config values
 */
const config = {
  URL_PATTERN: {
    CATEGORY_LANDING: '/c',
    PRODUCT_LIST: '/p',
    OUTFIT_DETAILS: '/outfit/',
    SEARCH_DETAIL: '/search',
    BUNDLE_DETAIL: '/b',
    STORE_LOCATOR: '/store-locator',
    CUSTOMER_SELF_HELP: '/customer-help/',
  },
};

export default config;
