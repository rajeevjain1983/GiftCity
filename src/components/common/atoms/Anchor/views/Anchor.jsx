// 9fbef606107a605d69c0edbcd8029e5d 
import React from 'react';
import { PropTypes } from 'prop-types';
import Link from 'next/link';
import {
  buildUrl,
  getAsPathWithSlug,
  getMappedPageHref,
  triggerPostMessage,
  getAPIConfig,
} from '../../../../../utils';
import withStyles from '../../../hoc/withStyles';

import styles from '../Anchor.style';

const getAsLinkPath = (IsSlugPathAdded, asPath, incomingUrl) => {
  return IsSlugPathAdded ? asPath || incomingUrl : getAsPathWithSlug(asPath || incomingUrl);
};

/**
 * @param {object} props : Props for Anchor
 * @desc This is an anchor component. The prop anchorVariation determines the color of the anchor.
 * The values supported for anchorVariation are : primary, secondary, tertiary.
 * The prop fontSizeVariation determines the font size of the anchor.
 * The value supported for fontSizeVariation are : small, medium, large.
 * The values of these are picked up from the theme.
 * An additional prop noLink is added to support simple anchor or next's Link conditionally.
 */

const Anchor = ({
  children,
  to,
  asPath,
  className,
  scroll,
  noLink,
  handleLinkClick,
  shallow,
  title,
  target,
  url,
  text,
  dataLocator,
  IsSlugPathAdded,
  ...other
}) => {
  const targetVal = target || '_self';
  let incomingUrl = to || url;
  const asLinkPath = getAsLinkPath(IsSlugPathAdded, asPath, incomingUrl);
  if (!noLink) {
    incomingUrl = getMappedPageHref(incomingUrl);
  }
  const hrefUrl = asLinkPath || buildUrl(incomingUrl);

  const apiConfigObj = getAPIConfig();

  let AnchorComponent = null;
  if (children || text) {
    AnchorComponent = noLink ? (
      <a
        href={buildUrl(incomingUrl)}
        className={className}
        onClick={handleLinkClick}
        title={title}
        target={targetVal}
        data-locator={dataLocator}
        {...other}
      >
        {children}
      </a>
    ) : (
      <Link href={incomingUrl} as={asLinkPath} shallow={shallow} scroll={scroll}>
        <a
          className={className}
          href={hrefUrl}
          title={title}
          target={targetVal}
          {...(apiConfigObj.isAppChannel
            ? {
                onClick: () => triggerPostMessage(to || asLinkPath),
              }
            : '')}
          data-locator={dataLocator}
          {...other}
        >
          {children || text}
        </a>
      </Link>
    );
  }
  return AnchorComponent;
};

Anchor.propTypes = {
  children: PropTypes.string.isRequired,
  to: PropTypes.string.isRequired,
  asPath: PropTypes.string,
  scroll: PropTypes.string,
  className: PropTypes.string.isRequired,
  noLink: PropTypes.bool,
  handleLinkClick: PropTypes.func,
  shallow: PropTypes.bool,
  title: PropTypes.string,
  target: PropTypes.string,
  url: PropTypes.string,
  text: PropTypes.string,
  dataLocator: PropTypes.string,
  IsSlugPathAdded: PropTypes.bool,
};

Anchor.defaultProps = {
  asPath: '',
  scroll: '',
  noLink: false,
  handleLinkClick: () => {},
  shallow: false,
  title: '',
  target: '',
  url: '',
  text: '',
  dataLocator: '',
  IsSlugPathAdded: false,
};

export default withStyles(Anchor, styles);
export { Anchor as AnchorVanilla };
