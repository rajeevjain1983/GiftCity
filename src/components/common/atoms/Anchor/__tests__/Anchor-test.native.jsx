// 9fbef606107a605d69c0edbcd8029e5d 
import React from 'react';
import { shallow } from 'enzyme';
import { AnchorVanilla } from '../views/Anchor.native';

describe('Anchor Native', () => {
  let component;
  const navigate = jest.fn();
  let navigation;
  beforeEach(() => {
    navigation = {
      navigate,
      dispatch: jest.fn(),
    };
    component = shallow(
      <AnchorVanilla
        url="/p/Rainbow--The-Birthday-Girl--Graphic-Tee"
        navigation={navigation}
        text="click Me"
        internal
        name="Anchor"
      />
    );
  });

  it('should be defined', () => {
    expect(component).toBeDefined();
  });

  it('should render correctly', () => {
    expect(component).toMatchSnapshot();
  });

  it('should return abc component value one', () => {
    expect(component.find('[name="Anchor"]')).toHaveLength(1);
  });

  it('should call parseUrl', () => {
    component.setProps({ navigation });
    component.props().onPress();
    expect(navigate).toHaveBeenCalledTimes(1);
  });

  it('should call onPress for external url handler', () => {
    component.setProps({ external: true, onPress: navigate });
    component.props().onPress();
    expect(navigate).toHaveBeenCalledTimes(2);
  });

  it('should call for plp page', () => {
    component.setProps({ url: '/p/test' });
    component.props().onPress();
    expect(navigate).toHaveBeenCalledTimes(3);
  });

  it('should call for shop page', () => {
    component.setProps({ url: '/c/test' });
    component.props().onPress();
    expect(navigate).toHaveBeenCalledTimes(4);
  });

  it('should return null', () => {
    component.setProps({ url: '/test' });
    component.props().onPress();
    expect(navigate).toHaveBeenCalledTimes(4);
  });

  it('should return navigation to default browser', () => {
    component.setProps({ onPress: null });
    component.props().onPress();
    expect(navigate).toHaveBeenCalledTimes(5);
  });
});
