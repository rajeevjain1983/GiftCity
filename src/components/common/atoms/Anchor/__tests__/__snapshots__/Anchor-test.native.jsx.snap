// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Anchor Native should render correctly 1`] = `
<Styled(Component)
  accessibilityLabel="click Me"
  accessibilityRole="button"
  accessible={true}
  hitSlop={Object {}}
  justifyContent=""
  onPress={[Function]}
  style={Object {}}
>
  <Styled(Text)
    anchorVariation=""
    colorName={null}
    internal={true}
    margins={null}
    name="Anchor"
    navigation={
      Object {
        "dispatch": [MockFunction],
        "navigate": [MockFunction],
      }
    }
    url="/p/Rainbow--The-Birthday-Girl--Graphic-Tee"
  >
    click Me
  </Styled(Text)>
</Styled(Component)>
`;
